### DISCUSSION TITLE
GUI toolkit

### DISCUSSION THREAD

**Discussion by hinxx** [OpenBB-finance/OpenBB#258] [Ideas]
After asking this on reddit yesterday, I understand that the GUI is not the biggest thing as of today. Anyway, I would like to hear what would be your toolkit of choice once you get there. GTK, QT? Have you seen ImGui/ImPlot?

I was thinking on coding a (simple) python driven portfolio that would allow me to provide pretty much a subset of what you guys have here. I was looking into using ImGui since it is something I'm kind of familiar with and like the way how it can be quickly integrated in the exiting apps. It has a python biding that I'm using on my app. At the moment, ImPlot is missing the python binding, but I have made one myself last week for the purpose of plotting stonk data.

Before going into more details about why I feel bullish for ImGui/ImPlot, I would like to test the waters of what you guys see in store for GUI.

ImGui: https://github.com/ocornut/imgui
PyImGui: https://github.com/swistakm/pyimgui
ImPlot: https://github.com/epezent/implot

**Comment by DidierRLopes** [comment]
Hey @hinxx,

We are not yet considering working on a GUI for the terminal, we are focusing on adding new features in order to add value to the terminal. Along the way we are improving the codebase.

We would be more than happy to let you lead the way in implementing such GUI for the terminal, if this is something of your interest :)

I have personally never tried ImGui, but it looks really nice!

**Reply by hinxx** [reply]
I would be honored to contribute to this project as it looks very interesting to me. Seeing the daily activity from many contributors is also very promising for the quality of the code. 

I'll need to get familiar with the code you guys have here, first. Hope to come back with some demo UI to show you what to expect.

**Reply by DidierRLopes** [reply]
Amazing!! Can't wait to see the demo!!

We can have a call this week if you want, where I can introduce you to the code to speed up the process :)

**Reply by hinxx** [reply]
I'd be interested in a call. Can I contact you over the email seen on your profile?

**Reply by DidierRLopes** [reply]
Yes, feel free to do so! :) And we can chat from there!

**Comment by tilakpatel22** [comment]
@hinxx I have been working on Terminal with gui , I'm using dearpygui instead of imgui and the terminal is in core python - https://github.com/Fincept-Corporation/FinceptTerminal
